File: frontend/src/types/projectStatusPage.ts
import type { HealthCheckStatus } from "./healthCheck";

export interface ProjectStatusPage {
  project_id: number;
  enabled: boolean;
  slug: string | null;
  label: string | null;
  created_at: string | null;
  updated_at: string | null;
}

export interface ProjectStatusPageUpdateInput {
  enabled: boolean;
  label?: string | null;
}

export interface PublicStatusCheck {
  status: HealthCheckStatus;
  http_status_code: number | null;
  response_time_ms: number | null;
  checked_at: string;
}

export interface PublicStatusIncident {
  opened_at: string;
  last_observed_at: string;
  recovered_at: string | null;
}

export interface PublicStatusPage {
  label: string;
  target_url: string | null;
  current_status: HealthCheckStatus | null;
  last_checked_at: string | null;
  active_incident: PublicStatusIncident | null;
  history: PublicStatusCheck[];
  generated_at: string;
}

File: backend/app/schemas/project_status_page.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.health_check import HealthCheckStatus


class ProjectStatusPageUpdate(BaseModel):
    enabled: bool
    label: str | None = Field(default=None, max_length=200)


class ProjectStatusPageRead(BaseModel):
    project_id: int
    enabled: bool
    slug: str | None
    label: str | None
    created_at: datetime | None
    updated_at: datetime | None

    model_config = ConfigDict(from_attributes=True)


class PublicStatusCheckRead(BaseModel):
    status: HealthCheckStatus
    http_status_code: int | None
    response_time_ms: int | None
    checked_at: datetime

    model_config = ConfigDict(from_attributes=True)


class PublicStatusIncidentRead(BaseModel):
    opened_at: datetime
    last_observed_at: datetime
    recovered_at: datetime | None


class PublicStatusPageRead(BaseModel):
    label: str
    target_url: str | None
    current_status: HealthCheckStatus | None
    last_checked_at: datetime | None
    active_incident: PublicStatusIncidentRead | None
    history: list[PublicStatusCheckRead]
    generated_at: datetime
